akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala

```diff
@@ -107,6 +107,9 @@ trait ActorContext[T] {
    * given [[ActorRef]] terminates. This message is also sent when the watched actor
    * is on a node that has been removed from the cluster when using akka-cluster
    * or has been marked unreachable when using akka-remote directly.
+   *
+   * Will fail with an [[IllegalStateException]] if the same subject was watched before using `watchWith`.
+   * To change the termination message, unwatch first.
```

Comment:
Perhaps clarify that `watch` itself is idempotent for subsequent calls of same subject

Reply:
need another test for that?

Reply:
Good point. Done!